### PR TITLE
camera_calibration sample: fix writing of calibration settings. also update grammar.

### PR DESCRIPTION
**Pull request by paroj**

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
@paroj, please, fix mistakes http://pullrequest.opencv.org/buildbot/builders/precommit_docs/builds/2811 and there http://pullrequest.opencv.org/buildbot/builders/precommit_windows64/builds/2531/steps/compile%20release/logs/warnings%20%282%29

**Comment by mshabunin** [comment]
@paroj, as I understand the corresponding [tutorial](http://docs.opencv.org/ref/master/d4/d94/tutorial_camera_calibration.html) should be updated too ([source](https://github.com/Itseez/opencv/blob/master/doc/tutorials/calib3d/camera_calibration/camera_calibration.markdown)).

1. I'd suggest to replace all code blocks (`@code ... @endcode`) with the `@snippet` command for convenience ([the guide](http://docs.opencv.org/ref/master/d4/db1/tutorial_documentation.html#tutorial_documentation_commands_include)). 
    
    Mark snippet block in the source file:
    ```.cpp
//! [file_read]
...
//! [file_read]
```
Then refer it in the tutorial:
    ```
@snippet samples/cpp/tutorial_code/calib3d/camera_calibration/camera_calibration.cpp file_read
```

2. XML files inserted in the _Results_ section of the tutorial should be regenerated too (because tags have changed). Maybe it is better to use `out_camera_data.yml` file (or xml replacement) and doxygen `@include` command:
    ```
@include samples/cpp/tutorial_code/calib3d/camera_calibration/out_camera_data.yml
```

**Comment by paroj** [comment]
results can not be regenerated as the source images are not checked in. the checked in yml file is too verbose to be included as is.

File: samples/cpp/tutorial_code/calib3d/camera_calibration/camera_calibration.cpp [hunk 1]
```diff
@@ -245,32 +257,35 @@ int main(int argc, char* argv[])
     const Scalar RED(0,0,255), GREEN(0,255,0);
     const char ESC_KEY = 27;
 
-    for(int i = 0;;++i)
+    //! [get_input]
+    while(true)
```
Comment (mshabunin) [inline samples/cpp/tutorial_code/calib3d/camera_calibration/camera_calibration.cpp]: Looks like VS does not like this form of infinite loop definition :smiley: 
> ...\camera_calibration.cpp(261): warning C4127: conditional expression is constant

**Comment by mshabunin** [comment]
@paroj, thank you, it looks great! I have no other comments.
Can you, please, squash all commits and create another pull request from branch with different name (not _master_ or _2.4_, because our integration system can run into problems during the merge).

**Comment by mshabunin** [comment]
Moving to #3874

**Pull request closed without merging**